src/content.rs

```diff
@@ -204,14 +210,56 @@ impl Content {
             published,
         }
     }
+    pub fn load_shortcode_dir(
+        &mut self,
+        handlebar: &mut handlebars::Handlebars,
+    ) -> anyhow::Result<()> {
+        // TODO: rewrite all_files so we don't need to clone here.
+        let scripts = all_files(PathBuf::from(SHORTCODE_PATH))?;
+        for script in scripts {
+            // Relative file name without extension. Note that we skip any file
+            // that doesn't have this.
+            if let Some(fn_name) = script.file_stem() {
+                eprintln!(
+                    "shortcodes: registering {}",
+                    fn_name.to_str().unwrap_or("unknown")
+                );
+                handlebar
+                    .register_script_helper_file(&fn_name.to_string_lossy(), &script)
+                    .map_err(|e| anyhow::anyhow!("Shortcode {:?}: {}", &script, e))?;
+            }
+        }
+        Ok(())
+    }
 
     /// Render the body using a Markdown renderer
-    pub fn render_markdown(&self) -> String {
+    pub fn render_markdown(&mut self) -> String {
         let mut buf = String::new();
-
         // Might as well turn on all the lights on the Christmas tree
         let opt = markdown::Options::all();
-        let parser = markdown::Parser::new_ext(&self.body, opt).map(translate_relative_links);
+        let out: String;
+        let parser = if self.head.enable_shortcodes == Some(true) {
```

Comment:
nit: would this statement be simplified with a `match` statement instead of an `if let`?